packages/types/src/types/k8s/Secret.ts

```diff
@@ -0,0 +1,25 @@
+import { IoK8sApimachineryPkgApisMetaV1ObjectMeta } from '../../models';
+
+export interface Secret {
```

Comment:
this type is incorrect, it allow for combinations the go code [1] does not allow, with this type a user can create secrets that will not pass the go code validation, a way to fix that is to create separate secret types for each provider and then combine them, so each provider type allow different subset of this fields, see the original TS secret struct [2]

[1] https://github.com/kubev2v/forklift/blob/main/pkg/controller/provider/validation.go#L165
[2[ https://github.com/kubev2v/forklift-console-plugin/blob/main/packages/types/src/types/ProviderSecret.ts

Reply:
@yaacov Secret is a native k8s type (`core.Secret`). They are fetched via `/api/kubernetes/api/v1/namespaces/${namespace}/secrets/${secretName}`. Some of the secrets may contain our custom props and could be mapped back to our convenience types but this seems not guaranteed.

Reply:
> Secret is a native k8s type (core.Secret)

the secrets we create to work with providers should be verified see https://github.com/kubev2v/forklift-console-plugin/pull/418#discussion_r1214000057 , the go code verify them in several places in the controller code base to make sure the contain the correct fields [1]

> some of the secrets may contain our custom props and could be mapped back to our convenience types but this seems not guaranteed.

If you want to be able to fetch any k8s secret you should use a string to string dictionary, we want to use type safe structs to help use validate the types. while a api call can return any string to string dictionary, once we know it's a valid provider secret we need to trust the type to help us use it.

Having a lose type that can hold none valid secrets is not helpful, we want types that will be correct.

[1] https://github.com/kubev2v/forklift/blob/main/pkg/controller/provider/validation.go#L165

Reply:
@yaacov  I've reverted to string dictionary. The reason is that at this level we get base64 encoded values. Only after parsing they can be assigned to correct types i.e.:
```
// value in packages/mocks/src/definitions/basic/secrets.mock.ts
insecureSkipVerify: 'dHJ1ZQ==',

// definition in packages/types/src/types/secret/VSphereProviderSecret.ts 
insecureSkipVerify?: boolean;
```